bitbots_localization/src/localization.cpp

```diff
@@ -67,92 +53,124 @@ rcl_interfaces::msg::SetParametersResult Localization::onSetParameters(
   field_publisher_ = this->create_publisher<nav_msgs::msg::OccupancyGrid>(
       "field/map", rclcpp::QoS(rclcpp::KeepLast(1)).transient_local());
 
-  // Get field name
-  auto field = this->get_parameter("fieldname").as_string();
+  // Set the measurement timestamps to 0
+  line_pointcloud_relative_.header.stamp = rclcpp::Time(0);
+  goal_posts_relative_.header.stamp = rclcpp::Time(0);
+  fieldboundary_relative_.header.stamp = rclcpp::Time(0);
+
+  // Update all things that are dependent on the parameters and
+  // might need to be updated during runtime later if a parameter is changed
+  updateParams(true);
+
+  // Init the particles with the given distribution
+  RCLCPP_INFO(this->get_logger(), "Trying to initialize particle filter...");
+  reset_filter(config_.misc.init_mode);
+
+  // Init services that can be called from outside
+  reset_service_ = this->create_service<bl::srv::ResetFilter>(
+      "reset_localization", std::bind(&Localization::reset_filter_callback, this, _1, _2));
+  pause_service_ = this->create_service<bl::srv::SetPaused>(
+      "pause_localization", std::bind(&Localization::set_paused_callback, this, _1, _2));
+
+  // Start the timer that runs the filter
+  publishing_timer_ =
+      rclcpp::create_timer(this, this->get_clock(), rclcpp::Duration(0, uint32_t(1.0e9 / config_.particle_filter.rate)),
+                           std::bind(&Localization::run_filter_one_step, this));
+}
+
+// TODO handle parameter reloading
+void Localization::updateParams(bool force_reload) {
+  // Check if we don't need to update the parameters
+  if (!force_reload and !param_listener_.is_old(config_)) {
+    return;
+  }
+
+  RCLCPP_INFO(this->get_logger(), "Updating parameters...");
```

Comment:
I think, this should be debug

Reply:
I don't know tbh. I also thought that, but it is also quite useful to get some feedback if your reconfiguration actually worked. And it is a nice info to know that the parameters changed, so I think info is the best matching scope.

Reply:
This is especially true if you do it over the network